exercises/csharp/add-esdk-complete/src/Api.cs

```diff
@@ -0,0 +1,161 @@
+﻿using Amazon.DynamoDBv2;
+using Amazon.DynamoDBv2.Model;
+using Amazon.KeyManagementService;
+using Amazon.S3;
+using Amazon.S3.Model;
+using AWS.EncryptionSDK;
+using AWS.EncryptionSDK.Core;
+
+namespace DocumentBucket
+{
+    public class Api
+    {
+        private readonly AmazonDynamoDBClient amazonDynamoDBClient;
+        private readonly string tableName;
+        private readonly AmazonS3Client amazonS3Client;
+        private readonly string bucketName;
+        private readonly IKeyring keyring;
+        private readonly IAwsEncryptionSdk awsEncryptionSdk;
+
+        public Api(AmazonDynamoDBClient amazonDynamoDBClient, string tableName, AmazonS3Client amazonS3Client, string bucketName,
+            IAwsCryptographicMaterialProviders materialProviders)
+        {
+            this.amazonDynamoDBClient = amazonDynamoDBClient;
+            this.tableName = tableName;
+            this.amazonS3Client = amazonS3Client;
+            this.bucketName = bucketName;
+            keyring = materialProviders.CreateAwsKmsKeyring(new CreateAwsKmsKeyringInput
+            {
+                KmsKeyId = Config.FaytheCmkId,
+                KmsClient = new AmazonKeyManagementServiceClient()
+            }); ;
+
+            var esdkConfig = new AwsEncryptionSdkConfig
+            {
+                CommitmentPolicy = CommitmentPolicy.REQUIRE_ENCRYPT_ALLOW_DECRYPT,
+            };
+            awsEncryptionSdk = AwsEncryptionSdkFactory.CreateAwsEncryptionSdk(esdkConfig);
+        }
+
+        protected async Task<Dictionary<string, AttributeValue>> WriteItem<T>(T modeledItem) where T : BaseItem
+        {
+            Dictionary<string, AttributeValue> ddbItem = modeledItem.ToItem();
+            await amazonDynamoDBClient.PutItemAsync(tableName, ddbItem);
+            return ddbItem;
+        }
+
+        protected async Task<PointerItem> GetPointerItem(string key)
+        {
+            GetItemResponse response = await amazonDynamoDBClient.GetItemAsync(tableName, PointerItem.AtKey(key));
```

Comment:
nit: for brevity we've been following the convention of using `var` for all local variables, e.g. in the [.NET ESDK examples](https://github.com/aws/aws-encryption-sdk-dafny/tree/mainline/aws-encryption-sdk-net/Examples/Keyring)